src/wanderer.ts

```diff
@@ -46,165 +52,200 @@ export function digitizedMonstersRemaining(): number {
   },
   {
     zone: "Wormwood",
-    effect: $effect`Absinthe-Minded`,
-    potion: $item`tiny bottle of absinthe`,
+    available: () => have($effect`Absinthe-Minded`),
+    unlocker: $item`tiny bottle of absinthe`,
   },
   {
     zone: "RabbitHole",
-    effect: $effect`Down the Rabbit Hole`,
-    potion: $item`"DRINK ME" potion`,
+    available: () => have($effect`Down the Rabbit Hole`),
+    unlocker: $item`"DRINK ME" potion`,
+  },
+  {
+    zone: "Conspiracy Island",
+    available: () => airportAvailable("spooky"),
+    unlocker: $item`one-day ticket to Conspiracy Island`,
+    noInv: true,
+  },
+  {
+    zone: "Dinseylandfill",
+    available: () => airportAvailable("stench"),
+    unlocker: $item`one-day ticket to Dinseylandfill`,
+    noInv: true,
+  },
+  {
+    zone: "The Glaciest",
+    available: () => airportAvailable("cold"),
+    unlocker: $item`one-day ticket to The Glaciest`,
+    noInv: true,
+  },
+  {
+    zone: "Spring Break Beach",
+    available: () => airportAvailable("sleaze"),
+    unlocker: $item`one-day ticket to Spring Break Beach`,
+    noInv: true,
   },
 ];
 
-function acceptBestGuzzlrQuest() {
-  if (!Guzzlr.isQuestActive()) {
-    if (
-      Guzzlr.canPlatinum() &&
-      (!Guzzlr.haveFullPlatinumBonus() ||
-        (Guzzlr.haveFullBronzeBonus() && Guzzlr.haveFullGoldBonus()))
-    ) {
-      Guzzlr.acceptPlatinum();
-    } else if (Guzzlr.canGold() && (!Guzzlr.haveFullGoldBonus() || Guzzlr.haveFullBronzeBonus())) {
-      Guzzlr.acceptGold();
-    } else {
-      Guzzlr.acceptBronze();
-    }
-  }
-}
+function testZoneAndUsePotionToAccess(zone: Location | null) {
+  if (!zone) return false;
 
-function testZoneAndUsePotionToAccess() {
-  const guzzlZone = Guzzlr.getLocation();
-  if (!guzzlZone) return false;
-  const forbiddenZones: string[] = [""]; //can't stockpile these potions,
-  if (!get("_spookyAirportToday") && !get("spookyAirportAlways")) {
-    forbiddenZones.push("Conspiracy Island");
-  }
-  if (!get("_stenchAirportToday") && !get("stenchAirportAlways")) {
-    forbiddenZones.push("Dinseylandfill");
-  }
-  if (!get("_hotAirportToday") && !get("hotAirportAlways")) {
-    forbiddenZones.push("That 70s Volcano");
-  }
-  if (!get("_coldAirportToday") && !get("coldAirportAlways")) {
-    forbiddenZones.push("The Glaciest");
-  }
-  if (!get("_sleazeAirportToday") && !get("sleazeAirportAlways")) {
-    forbiddenZones.push("Spring Break Beach");
+  // these zones have equip or effect requirements that we don't want to deal with
+  const skiplist = $locations`The Oasis, The Bubblin' Caldera, Barrrney's Barrr, The F'c'le, The Poop Deck, Belowdecks, 8-Bit Realm, Madness Bakery, The Secret Government Laboratory`;
+  if (skiplist.includes(zone)) {
+    return false;
   }
-
-  zonePotions.forEach((place) => {
-    if (guzzlZone.zone === place.zone && !have(place.effect)) {
-      if (!have(place.potion)) {
-        buy(1, place.potion, 10000);
+  zoneUnlockers.forEach((place) => {
```

Comment:
In retrospect, it feels very silly to have this be a for each. Make it a `for` loop and `break` if you find the parent zone. I know I made it this way, but that was past me.